src/executor/mod.rs

```diff
@@ -0,0 +1,303 @@
+pub(crate) mod current_context;
+pub(crate) mod custom_executor;
+#[cfg(feature = "tokio")]
+pub(crate) mod spawn_blocking;
+
+use std::sync::OnceLock;
+
+use current_context::CurrentContextExecutor;
+use custom_executor::{CustomExecutor, CustomExecutorSyncClosure};
+
+use crate::{global_sync_strategy_builder, Error, ExecutorStrategy, GlobalStrategy};
+
+pub(crate) trait ExecuteSync {
+    /// Accepts a sync function and processes it to completion.
```

Comment:
I wonder if we should have `execute` (and in the future, `execute_async`) or something

Reply:
I feel like this doesn't need to be `execute_sync` since the `sync` is already implied by type of `F`

Reply:
Yeah, this is a vestige of when it was initially both supporting sync + async executors, I can cut it over